liquibase-core/src/main/java/liquibase/diff/output/changelog/DiffToChangeLog.java

```diff
@@ -124,15 +140,17 @@ public void print(String changeLogFile, ChangeLogSerializer changeLogSerializer)
                 database = determineDatabase(diffResult.getComparisonSnapshot());
             }
             newScopeObjects.put(DIFF_SNAPSHOT_DATABASE, database);
+            boolean finalFileExists = fileExists;
             Scope.child(newScopeObjects, new Scope.ScopedRunner() {
                 @Override
                 public void run() {
                     try {
-                        if (!file.exists()) {
+                        if (!finalFileExists) {
                             //print changeLog only if there are available changeSets to print instead of printing it always
-                            printNew(changeLogSerializer, file);
+                            printNew(changeLogSerializer, changeLogFile);
                         } else {
-                            Scope.getCurrentScope().getLog(getClass()).info(file + " exists, appending");
+                            Scope.getCurrentScope().getLog(getClass()).info(changeLogFile + " exists, appending");
+                            StringBuilder fileContents = new StringBuilder(StreamUtil.readStreamAsString(resource.openInputStream()));
```

Comment:
@nvoxland - is it okay that we're loading the changelog file into memory here? Would it be better to instead download the file from S3 to temp directory and continue to use the same `RandomAccessFile` logic that used to be here?

Reply:
With the new interfaces only exposing input/output streams, I think copying like this is all we can do. The files are probably generally small enough that streaming it to a temp file and then using RandomAccessFile is probably unneeded?

But I wouldn't be against doing it too. I suppose best now when we remember the code is in git history?

Reply:
One vote here for keeping the logic the way its written - loading the file into memory. The change will still be easily found in the git history when this is merged. I am not worried about that.